src/libraries/System.Private.CoreLib/src/System/IO/FileStream.cs

```diff
@@ -106,28 +59,9 @@ public FileStream(IntPtr handle, FileAccess access, bool ownsHandle, int bufferS
                 // safeHandle.SetHandleAsInvalid();
                 throw;
             }
-
-            // Note: Cleaner to set the following fields in ValidateAndInitFromHandle,
-            // but we can't as they're readonly.
-            _access = access;
-            _useAsyncIO = isAsync;
-
-            // As the handle was passed in, we must set the handle field at the very end to
-            // avoid the finalizer closing the handle when we throw errors.
-            _fileHandle = safeHandle;
-        }
-
-        public FileStream(SafeFileHandle handle, FileAccess access)
-            : this(handle, access, DefaultBufferSize)
-        {
         }
 
-        public FileStream(SafeFileHandle handle, FileAccess access, int bufferSize)
-            : this(handle, access, bufferSize, GetDefaultIsAsync(handle))
-        {
-        }
-
-        private void ValidateAndInitFromHandle(SafeFileHandle handle, FileAccess access, int bufferSize, bool isAsync)
+        private static void ValidateHandle(SafeFileHandle handle, FileAccess access, int bufferSize, bool isAsync)
```

Comment:
this would require making it a partial class (to `FileStreamHelpers.Unix.cs` and `FileStreamHelpers.Windows.cs` we would need to add `FileStreamHelpers.cs` with OS-agnostic helpers). I would prefer to avoid that for a single method and rather revisit it in the future when we have more OS-agnostic helpers.